src/main/scala/fly/play/jiraExceptionProcessor/JiraExceptionProcessor.scala

```diff
@@ -1,40 +1,108 @@
 package fly.play.jiraExceptionProcessor
 
-import play.api.mvc.RequestHeader
 import java.io.StringWriter
 import java.io.PrintWriter
-import play.api.mvc.Session
-import play.api.Play.current
-import java.security.MessageDigest
 import javax.mail.Message
+import net.kaliber.mailer.Email
+import net.kaliber.mailer.EmailAddress
+import net.kaliber.mailer.Mailer
+import net.kaliber.mailer.Session
+import net.kaliber.mailer.Recipient
+import play.api.Configuration
+import play.api.Logger
 import play.api.PlayException
-import play.Logger
-import play.api.libs.concurrent.Promise
-import play.modules.mailer.PlayConfiguration
-import play.api.libs.concurrent.Execution.Implicits._
+import play.api.libs.ws.WSClient
+import play.api.mvc.RequestHeader
 import scala.concurrent.Await
-import scala.concurrent.duration._
-import play.modules.mailer.Mailer
-import play.modules.mailer.Email
-import play.modules.mailer.EmailAddress
-import play.modules.mailer.Recipient
-import scala.language.postfixOps
+import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import scala.concurrent.duration._
 
-object JiraExceptionProcessor {
+class JiraExceptionProcessor(client: WSClient, configuration: Configuration)(implicit ec: ExecutionContext) {
 
-  def reportError(request: RequestHeader, ex: Throwable): Unit = {
+  // these are lazy so the configuration can have missing settings when disabled
+  private lazy val processorConfiguration = JiraExceptionProcessorConfiguration fromConfiguration configuration
+  private lazy val (jira, mailer) = (
+    new Jira(client, processorConfiguration),
+    new Mailer(Session fromConfiguration configuration)
+  )
+
+  /*
+   * Public functions do not return a Future. Implementers might forget
+   * to process the result of the future and with that fail to catch any
+   * reporting problems.
+   */
```

Comment:
Het is ook goed mogelijk dat er op dat punt geen future is om op in te haken (denk actors). Daarnaast is het essentieel dat je op de hoogte wordt gesteld dat het rapporteren van een error mis gaat.

Een bijkomende overweging is dat dit gebruikt wordt op het moment dat de applicatie niet meer kan recoveren (einde van de lijn) en daarmee heb je geen van de problemen die ontstaan door het wachten op futures.

Ik snap je probleem met het `Await` gebeuren en dat is in alle andere gevallen terecht. Ik ben er van overtuigd dat dit (naast het gebruik in tests) een valide gebruik is.